Dockerfile.dind

```diff
@@ -1,9 +1,35 @@
 FROM docker:1.12.3-experimental-dind
 
-RUN apk add --no-cache git tmux py-pip apache2-utils vim \
-    && pip install 'docker-compose==1.8.0'
+ENV DOCKER_STORAGE_DRIVER=overlay2
 
-COPY vimrc ./root/.vimrc
+RUN apk add --no-cache git tmux py-pip apache2-utils vim build-base gettext-dev curl
 
-CMD docker daemon --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 --storage-driver=vfs &>- & while true; do /bin/sh; done
+# Install Compose and Machine
+RUN pip install docker-compose==1.8.1
+RUN curl -L https://github.com/docker/machine/releases/download/v0.9.0-rc1/docker-machine-Linux-x86_64 \
+    -o /usr/bin/docker-machine && chmod +x /usr/bin/docker-machine
 
+# Compile and install httping
+# (used in orchestration workshop, and very useful anyway)
+RUN mkdir -p /opt && cd /opt && \
+    curl https://www.vanheusden.com/httping/httping-2.5.tgz | \
+    tar -zxf- && cd httping-2.5 && \
+    ./configure && make install LDFLAGS=-lintl
+
+# Replace modprobe with a no-op to get rid of spurious warnings
+# (note: we can't just symlink to /bin/true because it might be busybox)
+RUN rm /sbin/modprobe && echo '#!/bin/true' >/sbin/modprobe && chmod +x /sbin/modprobe
+
+# Install a nice vimrc file
+COPY vimrc /root/.vimrc
+
+# Move to our home
+WORKDIR /root
+
+# Remove IPv6 alias for localhost and start docker in the background ...
+CMD cat /etc/hosts >/etc/hosts.bak && \
+    sed 's/^::1.*//' /etc/hosts.bak > /etc/hosts && \
```

Comment:
`/etc/hosts` is a bind-mount, so `set -i` won't work, unfortunately. And it's done when the container is created, which is why I do it in `CMD` instead of `RUN`. Annoying :( But once we fix IPv6 support, this hack can go away.